src/components/pages/import.js

```diff
@@ -0,0 +1,158 @@
+import by from 'thenby';
+import Page from '../page.js';
+import PropTypes from 'prop-types';
+import urls from '../../utils/urls.js';
+import React, { useState } from 'react';
+import Logger from '../../common/logger.js';
+import { useTranslation } from 'react-i18next';
+import { Tooltip, Button, Table, Select, Spin } from 'antd';
+import { inject, useService } from '../container-context.js';
+import { DOCUMENT_IMPORT_TYPE } from '../../common/constants.js';
+import ImportApiClient from '../../services/import-api-client.js';
+import { useDateFormat, useLanguage } from '../language-context.js';
+import LanguageNameProvider from '../../data/language-name-provider.js';
+import CountryFlagAndName from '../localization/country-flag-and-name.js';
+import { CloudDownloadOutlined, CloudSyncOutlined } from '@ant-design/icons';
+
+const logger = new Logger(import.meta.url);
+
+function Import({ initialState, importApiClient }) {
+  const { t } = useTranslation('import');
+
+  const { language } = useLanguage();
+  const { formatDate } = useDateFormat();
+  const languageNameProvider = useService(LanguageNameProvider);
+
+  const sourceMenuItems = initialState.importSources;
+  const [selectedSource, setSelectedSource] = useState(null);
+  const [selectedDocumentKeys, setSelectedDocumentKeys] = useState([]);
+  const [importableDocuments, setImportableDocuments] = useState([]);
+
+  const handleImportClick = () => {
+    const tasks = selectedDocumentKeys
+      .map(key => importableDocuments.find(doc => doc.key === key))
+      .map(doc => ({
+        key: doc.key,
+        importedRevision: doc.importedRevision,
+        importableRevision: doc.importableRevision
+      }));
+    logger.info(`Dummy import sets tasks: ${JSON.stringify(tasks)}`);
+  };
+
+  const handleSourceMenuChange = async value => {
+    const newSelectedSource = sourceMenuItems.find(source => source.name === value);
+    setSelectedSource(newSelectedSource);
+    setImportableDocuments([]);
+    setSelectedDocumentKeys([]);
+
+    const { documents } = await importApiClient.getImports(newSelectedSource);
+    setImportableDocuments(documents);
+  };
+
+  const renderImportType = doc => {
+    let icon = doc.importType;
+    if (doc.importType === DOCUMENT_IMPORT_TYPE.add) {
+      icon = <CloudDownloadOutlined />;
+    }
+    if (doc.importType === DOCUMENT_IMPORT_TYPE.update) {
+      icon = <CloudSyncOutlined />;
+    }
+    return <Tooltip title={t(doc.importType)} className="ImportPage-importType">{icon}</Tooltip>;
+  };
+
+  const renderTitle = doc => {
+    if (!doc.slug) {
+      return <span>{doc.title}</span>;
+    }
+
+    const url = `${selectedSource.baseUrl}${urls.getArticleUrl(doc.slug)}`;
+    return <a href={url}>{doc.title}</a>;
+  };
+
+  const renderLanguage = doc => {
+    const langaugesData = languageNameProvider.getData(language);
+    const documentLanguageData = langaugesData[doc.language];
```

Comment:
This data is only available in en and de. Therefore it can only be used by the UI language (not the doc language). Also: typo "lang**au**gesData"

Reply:
I think I'm a bit confused then with the languages ( <- which even here I wrote langauges and then corrected 🤦‍♀️ ) as the docs page renders the doc language in the same manner.

Reply:
I see, then we should correct it also on the docs page. The `languagesData` contains labels for all languages in German and English to show the language name in the current UI language (which can be either en or de). But the document language can be anything.

Reply:
Oh no, it's correct, I'm the one that got confused. Please ignore.